sphinx/builders/html/__init__.py

```diff
@@ -490,11 +487,11 @@ def prepare_writing(self, docnames: Set[str]) -> None:
         rellinks: List[Tuple[str, str, str, str]] = []
         if self.use_index:
             rellinks.append(('genindex', _('General Index'), 'I', _('index')))
-        for indexname, indexcls, _content, _collapse in self.domain_indices:
+        for indexname, index in self.domain_indices:
```

Comment:
Before this change, `self.domain_indices` contains only domains having contents. On the other hand, it contains all domains. As a result, `rellinks` will contain hyperlinks to not existing pages. What I mentioned at https://github.com/sphinx-doc/sphinx/issues/10299#issuecomment-1084849148 is this. As you said, rellinks never use each index entry itself. But it implicitly checks the existence. So I consider this will break rellinks.

Reply:
You are right

My suggestion is to clean rellinks later on


https://github.com/sphinx-doc/sphinx/pull/10329/files#diff-be839d87d39d93093003a0a4ce40cf9610d0a9445e5e0aeb1cdc6bab527bfbbcR733

If this is not acceptable, I'll try to find an other way

Reply:
Modifying rellinks after generating HTML is too late. It does not change the HTML files already written. You need to rewrite them all again. This is why we generate domain index entries before writing.

Reply:
thank you @tk0miya for your help and patience.

I made an error when analysing when what code was called at what time, and I though that it was not too late to remove empty domains from rellinks.  I since understood why I (wrongly) ended up to this conclusion.

I am sorry for wasting your time with this.  This leads me to two improvement ideas I'll open issues for. Thank you again.